python/src/scipp/constants/__init__.py

```diff
@@ -0,0 +1,106 @@
+# flake8: noqa: E501
+r"""
+Physical and mathematical constants with units.
+This is a wrapper around `scipy.constants <https://docs.scipy.org/doc/scipy/reference/constants.html>`_.
+
+Mathematical constants:
+
+================  =================================================================
+``pi``            Pi
+``golden``        Golden ratio
+``golden_ratio``  Golden ratio
+================  =================================================================
+
+Physical constants:
+
+===========================  =============================================
+``c``                        speed of light in vacuum
+``speed_of_light``           speed of light in vacuum
+``mu_0``                     the magnetic constant :math:`\mu_0`
+``epsilon_0``                the electric constant (vacuum permittivity),
+                             :math:`\epsilon_0`
+``h``                        the Planck constant :math:`h`
+``Planck``                   the Planck constant :math:`h`
+``hbar``                     :math:`\hbar = h/(2\pi)`
+``G``                        Newtonian constant of gravitation
+``gravitational_constant``   Newtonian constant of gravitation
+``g``                        standard acceleration of gravity
+``e``                        elementary charge
+``elementary_charge``        elementary charge
+``R``                        molar gas constant
+``gas_constant``             molar gas constant
+``alpha``                    fine-structure constant
+``fine_structure``           fine-structure constant
+``N_A``                      Avogadro constant
+``Avogadro``                 Avogadro constant
+``k``                        Boltzmann constant
+``Boltzmann``                Boltzmann constant
+``sigma``                    Stefan-Boltzmann constant :math:`\sigma`
+``Stefan_Boltzmann``         Stefan-Boltzmann constant :math:`\sigma`
+``Wien``                     Wien displacement law constant
+``Rydberg``                  Rydberg constant
+``m_e``                      electron mass
+``electron_mass``            electron mass
+``m_p``                      proton mass
+``proton_mass``              proton mass
+``m_n``                      neutron mass
+``neutron_mass``             neutron mass
+===========================  =============================================
+
+In addition to the above variables, :mod:`scipp.constants` also contains the
+2018 CODATA recommended values [CODATA2018]_ database containing more physical
+constants.
+The database is accessed using :py:func:`scipp.constants.physical_constants`.
+
+.. [CODATA2018] CODATA Recommended Values of the Fundamental
+   Physical Constants 2018.
+   https://physics.nist.gov/cuu/Constants/
+"""
+import math as _math
+from .. import scalar, Variable
+
+
+def physical_constants(key: str, with_variance: bool = False) -> Variable:
```

Comment:
If we are already in the `constants` module, do we need to fully specify `physical_constants` here?
I thinking we could maybe have a shorter and more convenient (faster to type) name?
`sc.constants.get(...)` might be a bit short/non-descript but maybe something like that?

Reply:
I am using the same name is in `scipy`, so people familiar with `scipy` have an easier time.